src/components/monitorAvailabilityModal/hooks/useMonitorAvailabilityModal.ts

```diff
@@ -29,6 +29,48 @@ const useMonitorAvailabilityModal = () => {
   const [sameAvailability, setSameAvailability] =
     useState<TWeekDayAvailabilityState>(SAME_AVAILABILITY_INITIAL_STATE);
 
+  const availabilities: TAvailability[] = useMemo(() => {
+    const selectedWeekDays = weekDayAvailability.filter(
+      (weekDay) => weekDay.isSelected
+    );
+
+    const availability: TAvailability[] = selectedWeekDays.map((weekDay) => ({
+      weekDay: weekDay.weekDay,
+      hours: [
+        {
+          start: sameAvailability.isSelected
+            ? HOUR_OPTIONS[sameAvailability.fromHourIndex]
+            : HOUR_OPTIONS[weekDay.fromHourIndex],
+          end: sameAvailability.isSelected
+            ? HOUR_OPTIONS[sameAvailability.toHourIndex]
+            : HOUR_OPTIONS[weekDay.toHourIndex],
+        },
+      ],
+    }));
+    return availability;
+  }, [weekDayAvailability, sameAvailability]);
+
+  const dayIsSelected: boolean = useMemo(() => {
+    return availabilities.length ? true : false;
+  }, [availabilities]);
+
+  const hoursIsSelected: boolean = useMemo(() => {
+    let hourMissing = true;
+    if (dayIsSelected) {
+      availabilities.map((day) => {
+        if (
+          day.hours[0].start === undefined ||
+          day.hours[0].end === undefined
+        ) {
+          hourMissing = true;
+          return;
+        }
+        hourMissing = false;
+      });
+    }
+    return !hourMissing;
```

Comment:
A função ideal para iterar sobre uma lista quando não se quer alterar os elementos dela é a `forEach`. De qualquer forma, como nesse caso vc não precisa percorrer a lista inteira, o ideal seria assim:

```suggestion
    if (dayIsSelected) {
      for (let day of availabilities) {
        if (!day.hours[0].start || !day.hours[0].end) {
          return true;
        }
      }
    }
    return false;
```

Reply:
Do jeito que tá agora, se o último dia estiver preenchido, os de cima tanto fazem pq o botão vai ficar habilitado:

![Captura de Tela 2023-05-18 às 14 43 40](https://github.com/ajuda-ai-ufam/frontend/assets/26875510/b41cf97f-b313-453e-9674-bc222b844e91)